share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
Let me jump in here.
Here is a simple example taken from `pthread.c` that Monty wrote:
```
pthread_t pthread_self(void) {
  int tmp_tid = __VERIFIER_nondet_int();
  __SMACK_code("call @ := corral_getThreadID();", tmp_tid);
  // Print actual tid to SMACK traces
  int actual_tid = tmp_tid;
  return actual_tid;
}
```
So Monty uses `pthread_t` as a very convenient place to store a thread id, and hence he defined this type to be an integer. Now, how `pthread_t` is actually defined is implementation-dependent, and for some pthread implementations it is in fact a struct for example. If you wanna be really crazy you can probably happily define it to be a float for example. If such headers would be included, then Monty's code above would not work, right?

Now, this is actually related to a major design decision we had to make early on - should we write models in Boogie or in C. I thought that having models be written in C as much as possible is the way to go, and I still do. Are you of a different opinion?